src/pydargs/__init__.py

```diff
@@ -130,4 +146,15 @@ def _parse_datetime(date_string: str, is_date: bool, date_format: Optional[str]
     return result.date() if is_date else result
 
 
+def _parse_union(value: str, union_type: Type) -> Any:
+    for arg in get_args(union_type):
+        if isinstance(None, arg):
+            continue
+        try:
+            return arg(value)
```

Comment:
```suggestion
    for type_ in get_args(union_type):
        if isinstance(None, arg):
            continue
        try:
            return type_(value)
```
the results from `get_args` are the types we want to cast to, right? Like `str`, `int`. 
I think something like this `type_` (but feel free to replace it with something else _type, kind, ..) is a little bit easier to read 😄